### DISCUSSION TITLE
Custom code execution widget

### DISCUSSION THREAD

**Discussion by Harmanpreet-96** [Azure/api-management-developer-portal#2015] [General]
We are using the managed version of the developer portal and need a custom widget that would allow the execution of some code. Is there an option to run some custom code.

**Comment by cgorman-cu** [comment]
There is now.

You can either have Custom HTML code by adding a widget and selecting "Custom HTML code".

Or you can import a React, TypeScript, or Vue custom widget using VSCode and npm. Go to the "custom widgets" in editor mode then follow the commands to setup the scaffold.
